src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -189,6 +191,20 @@ public override void OnError(Exception ex)
                 TraceCompilationDiagnostics(exc.Diagnostics, LogTargets.User);
                 throw;
             }
+            catch (CompilerErrorException exc)
+            {
+                TraceOnPrimaryHost("Script compiler error", TraceLevel.Error, _compilationService.GetType().Name, exc);
```

Comment:
If these are sporadic, is it possible for one host to have an exception while the others don't? If that's the case, we shouldn't just be tracing from the primary host here, should we?